From 50733dfc7dfb7a227ce8d01cffe4ec4f4e92e161 Mon Sep 17 00:00:00 2001
From: Patrick Hunt <phunt1@gmail.com>
Date: Mon, 28 Feb 2011 13:18:23 -0800
Subject: [PATCH 13/22] CLOUDERA-BUILD. Add support for '+' in the version string.

Description: Update code generator to correctly handle our non-standard + format in the version string
Reason: Improvement
Author: Patrick Hunt
Ref:  ZOOKEEPER-913
---
 .../org/apache/zookeeper/version/util/VerGen.java  |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/java/main/org/apache/zookeeper/version/util/VerGen.java b/src/java/main/org/apache/zookeeper/version/util/VerGen.java
index be494cd..b8cd483 100644
--- a/src/java/main/org/apache/zookeeper/version/util/VerGen.java
+++ b/src/java/main/org/apache/zookeeper/version/util/VerGen.java
@@ -116,7 +116,7 @@ public class VerGen {
     public static Version parseVersionString(String input) {
         Version result = new Version();
 
-        Pattern p = Pattern.compile("^(\\d+).(\\d+).(\\d+)(-(.+))?$");
+        Pattern p = Pattern.compile("^(\\d+).(\\d+).(\\d+)([-+](.+))?$");
         Matcher m = p.matcher(input);
 
         if (!m.matches()) {
-- 
1.7.0.4

